src/dnx.mono.managed/EntryPoint.cs

```diff
@@ -69,82 +68,92 @@ public static int Main(string[] arguments)
             {
-                break;
+                ExpandArgument(arguments[i], expandedArgs);
             }
-
-            // Copy the option
-            expandedArgs.Add(arguments[pathArgIndex]);
-
-            // Copy the value if the option has one
-            if (optionValNum > 0 && (++pathArgIndex < arguments.Length))
+            else
             {
-                expandedArgs.Add(arguments[pathArgIndex]);
+                expandedArgs.Add(arguments[i]);
             }
         }
 
-        // No path argument was found, no expansion is needed
-        if (pathArgIndex >= arguments.Length)
-        {
-            return arguments;
-        }
+        return expandedArgs.ToArray();
+    }
 
-        // Start to expand appbase option from path
+    private static void ExpandArgument(string argument, List<string> expandedArgs)
+    {
         expandedArgs.Add("--appbase");
 
-        var pathArg = arguments[pathArgIndex];
-        if (pathArg.EndsWith(".exe", StringComparison.OrdinalIgnoreCase) ||
-            pathArg.EndsWith(".dll", StringComparison.OrdinalIgnoreCase))
+        if (argument.EndsWith(".dll", StringComparison.OrdinalIgnoreCase) ||
+            argument.EndsWith(".exe", StringComparison.OrdinalIgnoreCase))
         {
             // "dnx /path/App.dll arg1" --> "dnx --appbase /path/ /path/App.dll arg1"
             // "dnx /path/App.exe arg1" --> "dnx --appbase /path/ /path/App.exe arg1"
-            expandedArgs.Add(Path.GetDirectoryName(Path.GetFullPath(pathArg)));
-            expandedArgs.AddRange(arguments.Skip(pathArgIndex));
+            expandedArgs.Add(Path.GetDirectoryName(Path.GetFullPath(argument)));
+            expandedArgs.Add(argument);
+
+            return;
         }
-        else
+
+        if (argument.Equals(".", StringComparison.Ordinal))
+        {
+            // "dnx . run" --> "dnx --appbase . Microsoft.Framework.ApplicationHost run"
+            expandedArgs.Add(argument);
+            expandedArgs.Add("Microsoft.Framework.ApplicationHost");
+            return;
+        }
+
+        if (string.Equals(Path.GetFileName(argument), "project.json", StringComparison.OrdinalIgnoreCase))
         {
-            var fileName = Path.GetFileName(pathArg);
-            if (string.Equals(fileName, "project.json", StringComparison.OrdinalIgnoreCase))
+            expandedArgs.Add(Path.GetDirectoryName(Path.GetFullPath(argument)));
+            expandedArgs.Add("Microsoft.Framework.ApplicationHost");
+            return;
+        }
+
+        // "dnx run" --> "dnx --appbase . Microsoft.Framework.ApplicationHost run"
+        expandedArgs.Add(".");
+        expandedArgs.Add("Microsoft.Framework.ApplicationHost");
+        expandedArgs.Add(argument);
+    }
+
+    private static int FindAppBaseOrNonHostOption(string[] arguments)
+    {
+        for (var i = 0; i < arguments.Length; i++)
+        {
+            if (string.Equals(arguments[i], "--appbase", StringComparison.OrdinalIgnoreCase))
```

Comment:
It seems that in `CommandUtil` the options names are compared in case sensitive way. 
https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.CommandLineUtils.Sources/CommandLine/CommandLineApplication.cs#L125